AsyncNats/Util/NatsKey.cs

```diff
@@ -3,10 +3,29 @@
 namespace EightyDecibel.AsyncNats
 {
     using System;
-    using System.Collections.Generic;
     using System.Runtime.CompilerServices;
     using System.Text;
 
+    public readonly ref struct NatsInlineKey
```

Comment:
And how big is the advantage of a ref-struct vs the normal struct? [more: do we need the aditional NatsInlineKey struct?]

Reply:
At this level, the advantage is small, but not zero. It avoids a potential copy instruction when passing it around method calls.   It also has to be a ref readonly struct in order to contain a ReadOnlySpan<byte> instead of a ReadonlyMemory<byte> as the normal NatsKey does, allowing us to use stack allocated keys.